src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -1633,17 +1662,26 @@ def to_frame(self, index: bool = True, name: Hashable | None = None) -> DataFram
         #       0               100
         #       1               200
         #       2               300
+        # If `name` is specified, use it as new column name; otherwise, set new column name to the original index name.
+        # Note there is one exception case: when the original index name is None, the new column name should be 0.
+        if name != native_pd._libs.lib.no_default:
+            new_col_name = name
+        else:
+            new_col_name = self._query_compiler.get_index_name()
```

Comment:
hmm, i only see we provide get_index_names() methods in the snowflake query compiler, is this added somewhere else?

Reply:
It was in query_compiler. We can just use self.name instead.